staging/src/k8s.io/legacy-cloud-providers/azure/azure_vmss_cache.go

```diff
@@ -73,7 +73,8 @@ func (ss *scaleSet) newVMSSCache() (*azcache.TimedCache, error) {
 				return nil, rerr.Error()
 			}
 
-			for _, scaleSet := range allScaleSets {
+			for i := range allScaleSets {
+				scaleSet := allScaleSets[i]
```

Comment:
in this loop, `scaleSet ` does not change, it's read-only

Reply:
looks like you mean this code:
```
vmss:       &scaleSet,
```

Reply:
you are correct:
https://github.com/golang/go/wiki/CommonMistakes#using-reference-to-loop-iterator-variable

Reply:
yes, you get it.

Reply:
Good catch, thanks for the fix